test/parallel_api/algorithm/alg.modifying.operations/transform_binary.pass.cpp

```diff
@@ -47,9 +58,9 @@ check_and_reset(InputIterator1 first1, InputIterator1 last1, InputIterator2 firs
         // check
         Out expected = Out(1.5) + *first1 - *first2;
         Out actual = *out_first;
-        if (::std::is_floating_point_v<Out>)
+        if constexpr (::std::is_floating_point_v<Out>)
         {
-            EXPECT_TRUE((expected > actual ? expected - actual : actual - expected) < Out(1e-7),
+            EXPECT_TRUE((expected > actual ? expected - actual : actual - expected) < get_epsilon<Out>(),
```

Comment:
In regard to two floating point numbers comparison please have a look
https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/

Reply:
in result, the overloads of `expect_equal_val` for floating point numbers were added.